native/src/feature/onboarding/screen/VerifyPhoneNumberCode.tsx

```diff
@@ -1,10 +1,75 @@
+import { QueryVerifyPhoneNumberCodeArgs } from "@corecodeio/libraries/api";
+import { StackNavigationProp } from "@react-navigation/stack";
 import React from "react";
-import { View, Text } from "react-native";
+import { Text, TextInput } from "react-native";
+import { PrimaryButton } from "../../../common/component/button";
+import { View } from "../../../common/component/view";
+import { DependencyContext } from "../../../common/context/DependencyContext";
+import { OnboardingStackScreenName } from "../../../navigation/model/OnboardingStackScreenName";
+import { OnboardingStackParamList } from "../../../navigation/types/OnboardingStackParamList";
+import { OnboardingCodeInjectionKey } from "../InjectionKey";
 
-export const VerifyPhoneNumberCode: React.FC = () => (
-  <View>
-    <Text>CheckCode</Text>
-  </View>
-);
+type Props = {
+  navigation: StackNavigationProp<
+    OnboardingStackParamList,
+    OnboardingStackScreenName.VerifyPhoneNumberCode
+  >;
+};
+
+export const VerifyPhoneNumberCode: React.FC<Props> = ({ navigation }) => {
+  const dependencies = React.useContext(DependencyContext);
+  const onboarding = dependencies.provide(OnboardingCodeInjectionKey);
+
+  const [args, setInput] = React.useState<QueryVerifyPhoneNumberCodeArgs>({
+    input: {
+      phoneNumber: "",
+      code: "",
+    },
+  });
+
+  const {
+    executeVerifyPhoneNumberCode,
+    result,
+    queryResult: { error: VerifyPhoneNumberCodeError },
+  } = onboarding.useVerifyPhoneNumberCode();
+
+  React.useEffect(() => {
+    if (!Boolean(result) || !result.valueOf()) {
+      return;
+    }
+
+    navigation.navigate(OnboardingStackScreenName.VerifyPhoneNumberCode);
+  }, [result]);
+
+  const onSetVerificationCode = (code: string) => {
+    let phoneNumber = args.input.phoneNumber;
+    setInput({ input: { phoneNumber, code } });
+  };
```

Comment:
Cuando sean valores que no se reasignan dentro del scope, usemos `const`. 

Por otro lado,  no es necesaria esta `const phoneNumber` ya que no hacemos nada con ella después:

```suggestion
  const onSetVerificationCode = (code: string) => {
    setInput({ input: { phoneNumber: args.input.phoneNumber, code } });
  };
```